src/logic/basic.lean

```diff
@@ -391,6 +391,9 @@ variables {α : Sort*} {p q : α → Prop} {b : Prop}
 
 def Exists.imp := @exists_imp_exists
 
+lemma exists_imp_exists' {α β : Sort*} {p : α → Prop} {q : β → Prop} (f : α → β) (h : ∀ a, p a → q (f a)) (p : ∃ a, p a) : ∃ a, q a :=
```

Comment:
You have `a : α` and `a : β` in `p` and conclusion, respectively. Is it intentional? I'd rename one of them.

Reply:
Good point. I'll do that.